src/lib.rs

```diff
@@ -1,39 +1,52 @@
 extern crate openvr_sys;
 
 use std::sync::atomic::{Ordering, AtomicBool, ATOMIC_BOOL_INIT};
-use std::{fmt, error};
-use std::ffi::CStr;
+use std::{fmt, error, ptr, mem};
+use std::ffi::{CStr, CString};
+use std::cell::Cell;
 
 use openvr_sys as sys;
 
 mod tracking;
 
-mod system;
-mod compositor;
+pub mod system;
+pub mod compositor;
+pub mod render_models;
+pub mod property;
 
 pub use tracking::*;
 
+pub use sys::VkPhysicalDevice_T;
+pub use sys::VkDevice_T;
+pub use sys::VkInstance_T;
+pub use sys::VkQueue_T;
+
 static INITIALIZED: AtomicBool = ATOMIC_BOOL_INIT;
 
 /// Initialize OpenVR
 ///
 /// # Panics
+///
 /// When the library has already been initialized
-pub fn init(ty: ApplicationType) -> Result<Context, InitError> {
+///
+/// # Safety
+///
+/// The `Context` MUST be dropped or shut down with `Context::shutdown` before shutting down the graphics API.
+pub unsafe fn init(ty: ApplicationType) -> Result<Context, InitError> {
```

Comment:
Also here, I think it is in general a better idea to provide safe functions. 
Why is it unsafe anyway? The Context has Drop implemented and will be dropped automatically anyway.